adr/005-transformer-runtime-context.md

```
@@ -0,0 +1,259 @@
+   - Propagate any transformer error up — short-circuit the remaining field applications
+   - Return `(*ent.XCreate, error)` / `(*ent.XUpdate, error)` / `(*ent.XUpdateOne, error)` accordingly
+3. Update any internal call sites in generated helper code (upsert, bulk) to thread ctx through.
+4. Emit a CHANGELOG entry flagging the breaking signature change with before/after examples.
+
+### User migration
+
+One-time, compile-time-enforced:
+
+1. `go generate ./...`
+2. Update every `ApplyDomain*` call to pass `ctx` and handle the new error:
+   ```go
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)